src/bfx_tools/gatk.ml

```diff
@@ -639,3 +647,96 @@ let mutect2
     in
     let final_vcf = result_prefix ^ "-merged.vcf" in
     Vcftools.vcf_concat ~run_with targets ~final_vcf ~more_edges
+
+
+(** Combine variant records from different sources.
+
+    cf. https://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.php  *)
+let combine_variants
+    ?(configuration=Configuration.Combine_variants.default)
+    ?(edges=[])
+    ~run_with
+    ~reference_genome
+    ~variant_sources
+    ~output_vcf_path
+    ~genotype_merge_option
+  =
+  let open KEDSL in
+  let gatk = Machine.get_tool run_with Machine.Tool.Default.gatk in
+  let reference_fasta = Reference_genome.fasta reference_genome in
+  let variant_sources =
+    List.map variant_sources
+      ~f:(function
+        | `Vcf vcf -> sprintf "--variant %s" vcf#product#path
+        | `Vcf_named_sample (sample_name, vcf) ->
+          sprintf "--variant:%s %s" sample_name vcf#product#path) in
+  let merge_option =
+    "-genotypeMergeOptions " ^
+    match genotype_merge_option with
+    | `Uniquify -> "UNIQUIFY"
+    | `Prioritize samples ->
+      "PRIORITIZE --priority " ^ String.concat ~sep:"," samples
```

Comment:
> The priority list isn't required … (though I'm not sure what GATK does without it)

Me neither, and later in the docs it sounds as though it *is* required, or undefined behavior if it's missing… so I thought I'd just require it here.

> Also, shouldn't that be just one dash in -priority?

Good catch, thank you! GATK requires -- in some places and - in others… lots of fun.